[17기 박재현] 모델 작성

namujinju opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- 앱 생성 및 모델 작성

## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
- user, posting, order, product 앱 생성
- 각 앱별 모델 작성

## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
조건을 만족한다면 각 빈칸에 x 표시 해주세요
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

namujinju committed:
Add : models.py 작성
- user app
- product app
- posting app
- order app

namujinju committed:
Add : 빠진 모델 추가
- order app

namujinju committed:
Modify: models.py 수정
- 필드 unique=True 추가
- ManyToManyField 추가
- recipients 테이블 병합

soheon-lee commented on order/models.py:
```diff
@@ -0,0 +1,26 @@
+
+    class Meta:
+        db_table= 'orders'
+
+class OrderProduct(models.Model):
+    product_option = models.ForeignKey('product.ProductOption', on_delete=models.SET_NULL, null=True)
+    quantity       = models.IntegerField()
+    order          = models.ForeignKey('Order', on_delete=models.CASCADE)
+
```
goood!

soheon-lee commented on posting/models.py:
```diff
@@ -0,0 +1,66 @@
+    size       = models.ForeignKey('PostingSize', on_delete=models.CASCADE)
+    housing    = models.ForeignKey('PostingHousing', on_delete=models.CASCADE)
+    style      = models.ForeignKey('PostingStyle', on_delete=models.CASCADE)
+    space      = models.ForeignKey('PostingSpace', on_delete=models.CASCADE)
+    like_user  = models.ManyToManyField('user.User', through='PostingLike', related_name='posting_like_user')
+    scrap_user = models.ManyToManyField('user.User', through='PostingScrap', related_name='posting_scrap_user')
+
```
related_name 지정 잘 하셨어요!
그런데, 
`posting_like_user` 이라는 related_name이 지칭하는 의미가 .. 그러니까 like를 한 'user' 인가요, like를 한 'posting'인가요?
`posting_scrap_user` 이라는 이름이 지칭하는 의미는 scrap을 한 'user'인가요, scrap을 한 'posting'인가요?

chaehyun-dev replied:
posting을 like 하는 user
posting을 scrap 하는 user 의 의미입니다!!
좀더 의미가 명확하게 짓는게 좋을까요??

soheon-lee replied:
그렇다면 related_name 이름을 잘못지정하셨습니다 하핫
여기 붙이는 이름은 해당 클래스를 지칭해야합니다.
https://velog.io/@brighten_the_way/Django%EC%99%80-Reverse-relations%EA%B3%BC-Relatedname
여기를 한 번 참조해보세용 :)

chaehyun-dev replied:
넵 수정하였습니다!

soheon-lee commented on product/models.py:
```diff
@@ -0,0 +1,113 @@
+
+    class Meta:
+        db_table = 'delivery_periods'
+
+class DeliveryFee(models.Model):
+    won = models.IntegerField(unique=True)
+
+    class Meta:
+        db_table = 'delivery_fees'
```
```suggestion

class DeliveryFee(models.Model):
    won = models.IntegerField(unique=True)

    class Meta:
        db_table = 'delivery_fees'
```

`won` 보다는 `fee` 나 `price`가 적당한 속성이름일 것 같습니담!

chaehyun-dev replied:
넵! 좀더 '요금' 같이 생긴 fee 로 하겠습니다!

soheon-lee commented on product/models.py:
```diff
@@ -0,0 +1,113 @@
+
+    class Meta:
+        db_table = 'delivery_fees'
+
+class DeliveryMethod(models.Model):
+    name = models.CharField(max_length=45, unique=True)
+
+    class Meta:
+        db_table = 'delivery_methods'
```
```suggestion

class DeliveryType(models.Model):
    name = models.CharField(max_length=45, unique=True)

    class Meta:
        db_table = 'delivery_types'
```
`method`  보다는 `type` 의미상 더 적절하다고 판단됩니다 :)

chaehyun-dev replied:
넵!!!!! 감사합니다 ~.~

soheon-lee commented:
복잡한 모델 설계하시느라 너무너무 고생하셨습니다 !

namujinju committed:
Modify: 테이블, 필드, 속성값 수정사항 반영
- related_name의 의미를 더 명확하게 수정
- DeliveryFee 필드 이름을 won에서 fee로 변경
- DeliveryMethod 클래스 이름을 DeliveryType로 변경

JELLO-Kim commented:
알람..확인용...

namujinju committed:
Fix: 테이블 참조 오류 수정
- 클래스를 DeliveryType으로 바꾸면서 참조 이름 오류가 생겼던 부분 수정
- migration 파일 생성

JELLO-Kim commented:
소헌님.. 이것만 고치면 다음단계 가도 되나요?

namujinju committed:
Modify: related_name 변경
- ProductReview 클래스의 related_name을 user_like_review로 변경

namujinju committed:
Modify: related_name 이름 수정
- Posting 테이블의 related_name 수정

soheon-lee commented on product/models.py:
```diff
@@ -0,0 +1,113 @@
+    content    = models.CharField(max_length=255)
+    image_url  = models.URLField(max_length=2000, null=True)
+    rate       = models.IntegerField()
+    created_at = models.DateTimeField(auto_now_add=True)
+    like_user  = models.ManyToManyField('user.User', through='ReviewLike', related_name='user_like_review')
+
```
오 여기도 바로 고치셨군요 !! ㅎㅎ good good

soheon-lee commented on user/models.py:
```diff
@@ -0,0 +1,20 @@
+    
+    class Meta:
+        db_table = 'users'
+
+class Follow(models.Model):
+    follower  = models.ForeignKey('User', on_delete=models.CASCADE, related_name='follower')
+    following = models.ForeignKey('User', on_delete=models.CASCADE, related_name='following')
+
+    class Meta:
+        db_table='follows'
```
여기도 related_name 한 번 보셔야할 것 같아요 ㅎㅎ 원래 의도하신 바가 맞나요?

chaehyun-dev replied:
소헌님 정말.. 엄청난 부분을 일깨워주셨습니다.. 얼른 수정하고 push 하겠습니다!!

namujinju replied:
수정했습니다!

namujinju committed:
Modify: Follow 테이블 related_name 수정
- 필드 이름을 방향성이 명확하게 드러나게 수정
- related_name을 관계의 의미에 맞게 수정

namujinju committed:
Modify: field name 수정
- delivery_fees 테이블 필드명 수정
- convention 수정

namujinju committed:
Modify: SubCategory, DetailCategory 테이블 unique 제거

soheon-lee commented:
LGTM!
